lms/djangoapps/verified_track_content/tests/test_models.py

```diff
@@ -35,3 +46,140 @@ def test_unicode(self):
         config = VerifiedTrackCohortedCourse.objects.create(course_key=course_key, enabled=True)
         config.save()
         self.assertEqual(unicode(config), "Course: {}, enabled: True".format(self.SAMPLE_COURSE))
+
+
+class TestMoveToVerified(SharedModuleStoreTestCase):
+    """ Tests for the post-save listener. """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMoveToVerified, cls).setUpClass()
+        cls.course = CourseFactory.create()
+
+    def setUp(self):
+        self.user = UserFactory()
+        # Spy on number of calls to celery task.
+        celery_task_patcher = patch.object(
+            sync_cohort_with_mode, 'apply_async',
+            mock.Mock(wraps=sync_cohort_with_mode.apply_async)
+        )
+        self.mocked_celery_task = celery_task_patcher.start()
+        self.addCleanup(celery_task_patcher.stop)
+
+    def _enable_cohorting(self):
+        set_course_cohort_settings(self.course.id, is_cohorted=True)
+
+    def _create_verified_cohort(self):
+        add_cohort(self.course.id, VERIFIED_COHORT_NAME, CourseCohort.MANUAL)
+
+    def _enable_verified_track_cohorting(self):
+        """ Enable verified track cohorting for the default course. """
+        config = VerifiedTrackCohortedCourse.objects.create(course_key=self.course.id, enabled=True)
+        config.save()
+
+    def _enroll_in_course(self):
+        self.enrollment = CourseEnrollmentFactory(course_id=self.course.id, user=self.user)
+
+    def _upgrade_to_verified(self):
+        """ Upgrade the default enrollment to verified. """
+        self.enrollment.update_enrollment(mode=CourseMode.VERIFIED)
+
+    def _verify_no_automatic_cohorting(self):
+        self._enroll_in_course()
+        self.assertIsNone(get_cohort(self.user, self.course.id, assign=False))
+        self._upgrade_to_verified()
+        self.assertIsNone(get_cohort(self.user, self.course.id, assign=False))
+        self.assertEqual(0, self.mocked_celery_task.call_count)
+
+    def _unenroll(self):
+        self.enrollment.unenroll(self.user, self.course.id)
+
+    def _reenroll(self):
+        self.enrollment.activate()
+        self.enrollment.change_mode(CourseMode.AUDIT)
+
+    @mock.patch('verified_track_content.models.log.error')
+    def test_automatic_cohorting_disabled(self, error_logger):
+        """
+        If the VerifiedTrackCohortedCourse feature is disabled for a course, enrollment does not move
+        them into a cohort.
```

Comment:
Nit: `s/them/the learner` in these 3 method docstrings, to make it clear who is being placed into the cohort.